dstep/translator/MacroDefinition.d

```diff
@@ -1188,7 +1221,417 @@ Expression parseCondExpr(ref TokenRange tokens, bool[string] table)
+        return true;
+    }
+
+    if (specifiers.contains("double"))
+    {
+        if (specifiers.contains("long"))
+        {
+            if (specifiers.length != 2)
+                return false;
+
+            type = Type(CXTypeKind.CXType_LongDouble, "long double");
+            return true;
+        }
+
+        if (specifiers.length != 1)
+            return false;
+
+        type = Type(CXTypeKind.CXType_Double, "double");
+        return true;
+    }
+
+    if ((specifiers.contains("signed") && specifiers.contains("unsigned")) ||
+        (specifiers.contains("char") && specifiers.contains("short")) ||
+        (specifiers.contains("char") && specifiers.contains("long")) ||
+        (specifiers.contains("char") && specifiers.contains("__llong")) ||
+        (specifiers.contains("short") && specifiers.contains("long")) ||
+        (specifiers.contains("short") && specifiers.contains("__llong")))
+        return false;
+
+    if (specifiers.contains("char"))
+    {
+        if (specifiers.contains("signed"))
+        {
+            if (specifiers.length != 2)
+                return false;
+
+            type = Type(CXTypeKind.CXType_SChar, "signed char");
+        }
+        else if (specifiers.contains("unsigned"))
+        {
+            if (specifiers.length != 2)
+                return false;
+
+            type = Type(CXTypeKind.CXType_UChar, "unsigned char");
+        }
+        else
+        {
+            if (specifiers.length != 1)
+                return false;
+
+            type = Type(CXTypeKind.CXType_Char_S, "char");
+        }
+
+        return true;
+    }
+
+    if (specifiers.contains("short"))
+    {
+        if (specifiers.contains("unsigned"))
+            type = Type(CXTypeKind.CXType_UShort, "unsigned short");
+        else
+            type = Type(CXTypeKind.CXType_Short, "short");
+
+        return true;
+    }
+
+    if (specifiers.contains("__llong"))
+    {
+        if (specifiers.contains("unsigned"))
+            type = Type(CXTypeKind.CXType_ULongLong, "unsigned long long");
+        else
+            type = Type(CXTypeKind.CXType_LongLong, "long long");
+
+        return true;
+    }
+
+    if (specifiers.contains("long"))
+    {
+        if (specifiers.contains("unsigned"))
+            type = Type(CXTypeKind.CXType_ULong, "unsigned long");
+        else
+            type = Type(CXTypeKind.CXType_Long, "long");
+
+        return true;
+    }
+
+    if (specifiers.contains("int"))
+    {
+        if (specifiers.contains("unsigned"))
+            type = Type(CXTypeKind.CXType_UInt, "unsigned int");
+        else
+            type = Type(CXTypeKind.CXType_Int, "int");
+
+        return true;
+    }
+
+    if (specifiers.contains("unsigned"))
+    {
+        type = Type(CXTypeKind.CXType_UInt, "unsigned int");
+        return true;
+    }
+
+    if (specifiers.contains("signed"))
+    {
+        type = Type(CXTypeKind.CXType_Int, "int");
+        return true;
+    }
+
+    return false;
+}
```

Comment:
Eh, that link isn't very good, the tests are in unit_tests/MacroParsingTests.d:128 .